Memories/Memories/VCs/AddViewController.swift

```diff
@@ -0,0 +1,159 @@
+        self.present(actionSheet, animated: true)
+    }
+    
+    @IBAction func filterImage(_ sender: Any) {
+        filterButton.isSelected.toggle()
+        if filterButton.isSelected {
+        guard let image = originalImage else { return }
+        var ciImage = CIImage(data: image)
+        let filter = CIFilter.colorMonochrome()
+        filter.inputImage = ciImage
+        filter.color = CIColor(color: .black)
+        ciImage = filter.outputImage
+        guard let outputCI = ciImage else { return }
+        guard let outputCG = context.createCGImage(outputCI, from: CGRect(origin: CGPoint.zero, size: imageView.image!.size)) else { return }
+        let uiImage = UIImage(cgImage: outputCG)
+        imageView.image = uiImage
+            self.image = uiImage.pngData()
+        }
+        else {
+            imageView.image = UIImage(data: originalImage!)
+        }
+    }
+    
+    @IBAction func done(_ sender: Any) {
+        save()
+    }
+    
+    private func save() {
+        //Check for title
+        guard let title = titleField.text, !(titleField.text?.isEmpty ?? true) else { Helper.alert(on: self, "Please add a title", "You can't save this without one"); return }
+        titleText = title
+        //Get location
+        locationManager.delegate = self
+        locationManager.requestWhenInUseAuthorization()
+        guard CLLocationManager.locationServicesEnabled() else {
+            Helper.alert(on: self, "Please turn on access for your location", "We are unable to display this data to you otherwise"); return }
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        locationManager.startUpdatingLocation()
+    }
+    
+    private func finishSaving() {
+        guard let coordinates = coordinates else { return }
+        var memories = Helper.getMemories()
+        let image = filterButton.isSelected ? self.image : originalImage
+        memories.append(Memory(title: titleText!, image: image, videoURL: video, audioURL: audio, latitude: coordinates.latitude, longitude: coordinates.longitude))
+        print(memories)
+        Helper.setMemories(memories)
+        NotificationCenter.default.post(name: NSNotification.Name("NewMemory"), object: nil)
+        dismiss(animated: true, completion: nil)
+    }
+    
+    //MARK: Audio & Video
+    @IBAction func addAudio(_ sender: Any) {
+        audioButton.isSelected.toggle()
+        hideAudio(!audioButton.isSelected)
+        hideVideo(audioButton.isSelected)
+    }
+    
+    @IBAction func addVideo(_ sender: Any) {
+        videoButton.isSelected.toggle()
+        hideAudio(videoButton.isSelected)
+        hideVideo(!videoButton.isSelected)
+    }
+    
+    private func hideAudio(_ bool: Bool) {
+        audioButton.isSelected = !bool
+        audioView.isHidden = bool
+    }
+    
+    private func hideVideo(_ bool: Bool) {
+        videoButton.isSelected = !bool
+        videoView.isHidden = bool
+    }
```

Comment:
This logic doesn't quite let me hide both again. I can't get back to original state.

It was confusing to me that it expanded the other.